packages/server-core/src/media/file-browser/file-browser.class.ts

```diff
@@ -79,18 +80,32 @@ export class FileBrowserService implements ServiceMethods<any> {
   async get(directory: string, params?: UserParams): Promise<Paginated<FileContentType>> {
     if (!params) params = {}
     if (!params.query) params.query = {}
-    const { $skip, $limit, storageProviderName } = params.query
+    const { $recursive, $skip, $limit, storageProviderName } = params.query
 
     delete params.query.storageProviderName
     const skip = $skip ? $skip : 0
     const limit = $limit ? $limit : 100
+    const recursive = $recursive
 
     const storageProvider = getStorageProvider(storageProviderName)
     const isAdmin = params.user && params.user?.scopes?.find((scope) => scope.type === 'admin:admin')
     if (directory[0] === '/') directory = directory.slice(1) // remove leading slash
     if (params.provider && !isAdmin && directory !== '' && !/^projects/.test(directory))
       throw new Forbidden('Not allowed to access that directory')
     let result = await storageProvider.listFolderContent(directory)
+    let resultClone = [] as FileContentType[]
```

Comment:
this will only fetch results one more layer, projects can be nested deeoer.

instead replace L#95 with `let result = await storageProvider.listObjects(directory, recursive)`

Reply:
Ah I didn't realize it was that simple. To get deeper layers I was using the loop below.

Reply:
Wow. I didn't realize it was that simple. Here I wrote an entire loop to manually go through those directories and push them.